pkg/stanza/fileconsumer/scanner.go

```diff
@@ -35,11 +37,22 @@ func NewPositionalScanner(r io.Reader, maxLogSize int, startOffset int64, splitF
 		Scanner: bufio.NewScanner(r),
 	}
 
-	buf := make([]byte, 0, 16384)
+	bufSize := defaultBufSize
+	if maxLogSize < bufSize {
+		bufSize = maxLogSize
+	}
+
+	buf := make([]byte, 0, bufSize)
 	ps.Scanner.Buffer(buf, maxLogSize)
 
 	scanFunc := func(data []byte, atEOF bool) (advance int, token []byte, err error) {
 		advance, token, err = splitFunc(data, atEOF)
+		if (advance == 0 && token == nil && err == nil) && len(data) >= maxLogSize {
+			// reference: https://pkg.go.dev/bufio#SplitFunc
+			// splitFunc returns (0, nil, nil) to signal the Scanner to read more data but the buffer is full.
```

Comment:
We use [custom split functions](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/951245d4273845de59db3b00f5445bcad9ce4372/pkg/stanza/operator/helper/multiline.go#L52) in most cases, but I believe we are respecting this pattern.